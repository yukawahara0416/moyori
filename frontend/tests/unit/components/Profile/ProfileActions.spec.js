import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Profile/ProfileActions.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let propsData
let store
let auth

beforeEach(() => {
  propsData = {
    id: 1,
    user: { data: { id: 1 }, posts: [{ data: { id: 1 } }] },
    headers: { uid: 'tester@example.com' },
    currentUser: { data: { id: 1 } }
  }

  auth = {
    getters: {
      isLoggingIn: () => true
    }
  }

  store = new Vuex.Store({
    modules: {
      auth
    }
  })

  wrapper = shallowMount(Component, {
    localVue,
    propsData,
    store
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('id', () => {
    expect(wrapper.props().id).toStrictEqual(propsData.id)
    expect(typeof wrapper.vm.$props.id).toBe('number')
  })
  it('user', () => {
    expect(wrapper.props().user).toStrictEqual(propsData.user)
    expect(wrapper.props().user instanceof Object).toBe(true)
  })
  it('headers', () => {
    expect(wrapper.props().headers).toStrictEqual(propsData.headers)
    expect(wrapper.props().headers instanceof Object).toBe(true)
  })
  it('currentUser', () => {
    expect(wrapper.props().currentUser).toStrictEqual(propsData.currentUser)
    expect(wrapper.props().currentUser instanceof Object).toBe(true)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
