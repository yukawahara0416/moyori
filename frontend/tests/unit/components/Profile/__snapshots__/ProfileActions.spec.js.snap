// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-col-stub
  class="mb-5"
  cols="8"
  tag="div"
>
  <v-row-stub
    align="center"
    justify="center"
    no-gutter=""
    tag="div"
  >
    <profile-actions-items-stub
      user="[object Object]"
    />
     
    <profile-actions-buttons-stub
      currentuser="[object Object]"
      headers="[object Object]"
      id="1"
      user="[object Object]"
    />
  </v-row-stub>
</v-col-stub>
`;
